tree/tree/src/TBranch.cxx

```diff
@@ -1926,19 +1940,19 @@ TBasket* TBranch::GetFreshBasket(Int_t basketnumber, TBuffer* user_buffer)
 #endif
                fTree->AddAllocationCount(basket->GetResetAllocationCount());
             } else {
-               basket = fTree->CreateBasket(this);
+               basket = CreateOrReuseBasket();
```

Comment:
This change (and the 3 similar changes below) actually alter the behavior (here there is no consideration of the `user_buffer` (and/or old line 1943,1944,1945 should be removed).

This is likely the cause of the test failures.

I will continue testing tomorrow.  There are also additional change to be done to reduce memory churn (as is the `TBufferFile` actually end up deleting their data buffer making the re-use of basket relatively unimpactful).

Reply:
> This change (and the 3 similar changes below) actually alter the behavior (here there is no consideration of the `user_buffer` (and/or old line 1943,1944,1945 should be removed).

I have removed the change, actually I am not sure it's actually needed for my usecase.